added recipe for jam1015/evil-god-toggle

jam1015 opened the pull request:
makes god-mode a state of evil

### Brief summary of what the package does

This package provides a simple global minor‐mode for Emacs that lets you switch seamlessly between Evil’s modal (Vim-style) editing and God mode’s single-key “command” input. It defines three toggleable states:

    god: enter persistent God mode

    god-once: enter God mode for exactly one command, then automatically revert to Evil

    god-off: explicitly disable God mode and return to Evil

Under the hood it manages hooks to preserve or restore visual selections, remembers your previous Evil state, and offers customization for whether God mode is applied buffer-locally or globally, and how visual regions should persist across toggles.

### Direct link to the package repository

https://github.com/jam1015/evil-god-toggle

### Your association with the package

I'm the author and maintainer.
### Relevant communications with the upstream package maintainer

**None needed**
### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass -

⸺ `evil-god-toggle.el` with byte-compile using Emacs 30.1:
```
evil-god-toggle.el:88:11: Warning: assignment to free variable `evil-god-toggle--last-command'
In evil-god-toggle-execute-in-god-state:
evil-god-toggle.el:95:9: Warning: assignment to free variable `evil-god-toggle--last-command'
In evil-god-toggle-stop-choose-state:
evil-god-toggle.el:120:34: Warning: reference to free variable `evil-god-toggle-persist-visual'
evil-god-toggle.el:128:15: Warning: assignment to free variable `evil-god-toggle--visual-beg'
evil-god-toggle.el:129:15: Warning: assignment to free variable `evil-god-toggle--visual-end'
evil-god-toggle.el:130:15: Warning: assignment to free variable `evil-god-toggle--visual-forward'
In evil-god-toggle--exit-once:
evil-god-toggle.el:393:2: Warning: docstring wider than 80 characters
```

⸺ `evil-god-toggle.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- evil-god-toggle.el#L254: Loading a package should rarely add hooks
```
(i.e. how does a user undo this?)

jam1015 commented:
Ok I made PRs that addressed the compiler/melpazoid warnings.  I added the hooks to the body of the definition of the minor-mode that this package defines, and I moved variable definitions and defcustoms to the top of the file.

riscy commented:
Great!  Thank you for accommodating - I'll merge this now.

riscy commented:
And welcome to MELPA. :)

jam1015 commented:
thank you!
